### ISSUE TITLE
Stone Slab - Failing Survival Fly/ Passable

### ISSUE THREAD

**Issue by ChrisLane** [NoCheatPlus/Issues#337]
#### Short description of the issue.
When players jump onto a stone slab they fail the survival fly check. When players walk/run onto a stone slab they fail the passable check.

#### Complete output of the _ncp version_ command (please prefer text here).
---- Version information ----
#### Server ####
1.7.10-1614.58 ~MC: 1.7.10~
  detected: 1.7.10
#### NoCheatPlus ####
Plugin: 3.15.0-SNAPSHOT-sMD5NET-b1046
MCAccess: 1.4.6|1.4.7|1.5.x|1.6.x|1.7.x|1.8.x|1.9.x|? / Bukkit-API
Features:
  blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2
  checks: FastConsume | Gutenberg | AttackFrequency | FlyingFrequency | KeepAliveFrequency
  defaults: enforceLocation | storePlayers
  packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance | CatchAllAdapter
Hooks: AllViolations~NCP~ 1.0 | mcMMO~cncp~ 2.3
#### Related Plugins ####
CompatNoCheatPlus v6.6.3-SNAPSHOT-sMD5NET-b87 | ProtocolLib v3.7.0

#### How to reproduce the issue (is it random / always happens / side conditions)?
Always happens when a player moves onto a half slab.

#### Have you made any (noteworthy) changes to the default configuration of NCP?
None relating to passable or survival fly. CBReflect is turned off.

#### Do you run any other special plugins that affect game mechanics next to NC+
No.

#### Provide additional information, such as error/debug/violation logs.
NCP log for player failing checks: https://gist.github.com/ChrisLane/77ff25af1f3743397fab9d0a85cf04a8

The server is running on Thermos build 58 (https://github.com/CyberdyneCC/Thermos/releases/tag/58).

This issue was not a problem in the bukkit dev 1.7.10 release but has now been found with these recent versions.

**Comment by asofold** [comment]
With Thermos organizing minecraft code within packages, the reflection based module can't work (so can't the other modules). In general it may be possible to add support for block shapes with Thermos, but it currently is not the highest priority.

Technically, for those interested in (contributing to e.g.) a PR, perhaps one could add in auxiliary methods for getting classes, that will test several possible paths for specific classes to access - most straight forward would possibly be to test them all out within ReflectBase and use access methods located in ReflectBase to obtain full class names (e.g. getNMSClassName(classicSubPath), so internal mappings can be used for a quite simple change set. E.g. EntityPlayer -> entity.player.EntityPlayer or whatever is used extra to net.minecraft.server...). I'll probably  have a closer look at this some time soon-ish.

**Comment by ChrisLane** [comment]
@asofold Surely this shouldn't be happening for me if it's a reflection based issue since I have "cbreflect: false".

**Comment by asofold** [comment]
Oh the issue with reflection in this case is, that reflection isn't present :).

The reflection-based module is intended to bridge some gaps, where the server code doesn't change in a significant way, but the package naming has been changed for the server, aiming at minor updates of Minecraft + Spigot.

So in this case updating the reflection module would allow NCP to fetch block shapes properly. An alternative would be to write a dedicated compatibility module for Thermos, but that means depending on another server mod - not sure i want to do that. It could be made as a separate jar, perhaps.

I don't know how many blocks pose problems, there could be a fall back workaround, which will allow players (cheaters) to pass through such blocks and to stand on any height up to full block height. It'll also affect fall damage amounts and might lead to issues with continuous sprint+jump.

You would add to the section compatibility.blocks.overrideflags , e.g.:
STEP: solid+ground+xz100+height100+ground_height+ign_passable

This would let NCP assume the block to be full size and allow standing on any height (with few exceptions), and to pass through it (necessary to walk on lower than a block blocks).

**Comment by MyPictures** [comment]
Could reproduce it the Passable one and adding some logs here:
[PassableLogs.zip](https://github.com/NoCheatPlus/Issues/files/965369/PassableLogs.zip)
![screenshot](https://cloud.githubusercontent.com/assets/1159235/25544467/e30356d0-2c5a-11e7-9961-1aeed79fc328.png)

Also uploading a screenshot of how the structure looked like.

[21:36:42 INFO]: ---- Version information ----
[21:36:42 INFO]: #### Server ####
[21:36:42 INFO]: git-Spigot-a3f6ea5-a2af8f0 ~MC: 1.11.2~
[21:36:42 INFO]:   detected: 1.11.2
[21:36:42 INFO]: #### NoCheatPlus ####
[21:36:42 INFO]: Plugin: 3.15.2-SNAPSHOT-sMD5NET-b1088
[21:36:42 INFO]: MCAccess: 1.11-1.11.2 / Spigot-CB-1.11_R1
[21:36:42 INFO]: Features:
[21:36:42 INFO]:   blocks: BlocksMC1_4 | BlocksMC1_5 | BlocksMC1_6_1 | BlocksMC1_7_2 | BlocksMC1_8 | BlocksMC1_9 | BlocksMC1_10 | BlocksMC1_11
[21:36:42 INFO]:   checks: FastConsume | Gutenberg | HotFixFallingBlockPortalEnter | AttackFrequency | FlyingFrequency | KeepAliveFrequency
[21:36:42 INFO]:   defaults: pvpKnockBackVelocity
[21:36:42 INFO]:   packet-listeners: UseEntityAdapter | MovingFlying | OutgoingPosition | KeepAliveAdapter | SoundDistance | DebugAdapter
[21:36:42 INFO]: Hooks: AllViolations~NCP~ 1.0 | TestNCPHook 1.0
[21:36:42 INFO]: #### Related Plugins ####
[21:36:42 INFO]: ProtocolLib v4.2.1

**Comment by asofold** [comment]
I'll have to include this with upcoming passable fixes.

**Comment by asofold** [comment]
This hopefully is fixed in build 1098 - please someone confirm :) @MyPictures 

If it's due to not being able to use either of native or reflection, i'd postpone until issues arrise (1.12, ...)
